src/osd/ECCache.hpp

```diff
@@ -0,0 +1,273 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2016 Western Digital Corporation.
+ * Author: Tomy Cheru <tomy.cheru@sandisk.com>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef ECCACHE_HPP_INCLUDED
+#define ECCACHE_HPP_INCLUDED
+
+#include <iostream>
+//#include <string>
+#include <set>
+#include "include/buffer.h"
+#include "common/hobject.h"
+#include "common/Mutex.h"
+#include "common/Cond.h"
+
+class ecc_extent
+{
+  private:
+    unsigned eoff;
+    bufferlist edata;
+
+  public:
+    bool operator < (const ecc_extent& r) const
+    {
+      return eoff < r.eoff;
+    }
+
+    bool operator == (const ecc_extent& r) const
+    {
+      return eoff == r.eoff;
+    }
+
+    ecc_extent(unsigned o = 0,bufferlist d = {} )
```

Comment:
First, use a colon initializer list -- always.  Second, using claim() in a bufferlist you are passing by value is silly, pass it by const ref and use the copy constructor.